test/e2e/end_user_test.go

```diff
@@ -163,4 +135,99 @@ var _ = Describe("Openshift on Azure end user e2e tests [EndUser]", func() {
 		fmt.Println(result.Error().Error())
 		Expect(result.Error().Error()).To(ContainSubstring("pods \"sync-master-000000\" is forbidden: User \"enduser\" cannot get pods/log in the namespace \"kube-system\""))
 	})
+
+	It("should deploy a template with persistent storage and test failure modes", func() {
+		tpl := "django-psql-persistent"
+		By(fmt.Sprintf("instantiating the template and getting the route (%v)", time.Now()))
+		// instantiate the template
+		err := c.instantiateTemplate(tpl)
+		Expect(err).NotTo(HaveOccurred())
+		// Pull the route ingress from the namespace
+		route, err := c.rc.Routes(c.namespace).Get(tpl, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		// make sure only 1 ingress point is returned
+		Expect(len(route.Status.Ingress)).To(Equal(1))
+		host := route.Status.Ingress[0].Host
+		url := fmt.Sprintf("http://%s", host)
+
+		// Curl the endpoint and search for a string
+		httpc := &http.Client{
+			Timeout: 10 * time.Second,
+		}
+		var regex = regexp.MustCompile(`Page views:\s*(\d+)`)
+		By(fmt.Sprintf("hitting the route 3 times, expecting counter to increment (%v)", time.Now()))
+		var prevCounter, currCounter int
+		for i := 0; i < 3; i++ {
+			resp, err := httpc.Get(url)
+			Expect(err).NotTo(HaveOccurred())
+			defer resp.Body.Close()
+			Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+
+			contents, err := ioutil.ReadAll(resp.Body)
+			Expect(err).NotTo(HaveOccurred())
+			matches := regex.FindStringSubmatch(string(contents))
+			Expect(matches).NotTo(BeNil())
+
+			currCounter, err = strconv.Atoi(matches[1])
+			Expect(err).NotTo(HaveOccurred())
+			Expect(currCounter).Should(BeNumerically(">", prevCounter))
+			prevCounter = currCounter
+			time.Sleep(time.Second)
+		}
+
+		// Find the database deploymentconfig and scale to 0
+		var dcName = "postgresql"
+		for i := range []int{0, 1} {
```

Comment:
`for i := range []int32{0, 1}` better?

Reply:
I actually tried that before, but weirdly: https://play.golang.org/p/ar68WXvytUH